lms/static/scripts/frontend_apps/components/FilePickerFormFields.js

```diff
@@ -0,0 +1,90 @@
+import { Fragment, createElement } from 'preact';
+
+import {
+  contentItemForUrl,
+  contentItemForLmsFile,
+  contentItemForVitalSourceBook,
+} from '../utils/content-item';
+
+/**
+ * @typedef {import('./FilePickerApp').Content} Content
+ */
+
+/**
+ * @typedef FilePickerFormFields
+ * @prop {string} ltiLaunchURL - URL that the LMS should use to launch the
+ *   assignment.
+ * @prop {Content} content - Content for the assignment
+ * @prop {Record<string,string>} formFields - Form fields provided by the backend
+ *   that should be included in the response without any changes
+ */
+
+/**
+ * Generate an LTI launch URL for a given piece of content.
+ *
+ * @param {string} ltiLaunchURL
+ * @param {Content|null} content
+ */
+function contentItemString(ltiLaunchURL, content) {
+  let contentItem = null;
+
+  switch (content?.type) {
+    case 'url':
+      contentItem = contentItemForUrl(ltiLaunchURL, content.url);
+      break;
+    case 'file':
+      contentItem = contentItemForLmsFile(ltiLaunchURL, content.file);
+      break;
+    case 'vitalsource': {
+      contentItem = contentItemForVitalSourceBook(
+        ltiLaunchURL,
+        content.bookID,
+        content.cfi
+      );
+    }
+  }
+  return JSON.stringify(contentItem);
+}
+
+/**
+ * Render the hidden form fields in the file picker form containing information
+ * about the selected assignment.
+ *
+ * This form may be used in two different scenarios:
+ *
+ *  - A "Content Item Selection" form when configuring an assignment. In this
+ *    case the form will look like "Section 3.2, Example Response" in
+ *    https://www.imsglobal.org/specs/lticiv1p0/specification
+ *  - When an assignment without any content configuration is launched.
+ *    See the `configure_module_item` view.
+ *
+ * @param {FilePickerFormFields} props
+ */
+export default function FilePickerFormFields({
+  content,
+  formFields,
+  ltiLaunchURL,
```

Comment:
Maybe instead of passing this `ltiLaunchIURL` around you could injected it here:

```
  const {  {filePicker: canvas: { ltiLaunchUrl }, }, } = useContext(Config);
```

Reply:
Same for formFields.

Reply:
Hmm ... I'm not sure about this. Currently all the configuration extraction related to this form is in one place, which is useful for comprehension.